File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        List<int[]> answer = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            answer.add(new int[i]);
        }
        
        int[] dy = new int[]{1, 0, -1};
        int[] dx = new int[]{0, 1, -1};
        
        int node = 1;
        int y = 0, x = 0, count = 0;
        
        while (count < n) {
            for (int i = 0; i < n - count; i++) {
                answer.get(y)[x] = node;
                node++;
                if (i != n - count - 1) {
                    y += dy[count % 3];
                    x += dx[count % 3];
                }
            }
            count++;
            y += dy[count % 3];
            x += dx[count % 3];
        }
        
        int[] result = new int[n * (n + 1) / 2];
        
        int j = 0;
        for (int[] a : answer) {
            for(int b : a) {
                result[j] = b;
                j++;
            }
        }
        
        
        
        return result;
    }
}

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    result = [[0] * i for i in range(1, n+1)]
    direction = [(1, 0), (0, 1), (-1,-1)]
    d = 0
    count = n
    i = 1
    y, x = -1, 0
    
    while count > 0:
        for round_count in range(count):
            y,x = y + direction[d][0], x + direction[d][1]
            result[y][x] = i
            i += 1
        count -= 1
        d = (d+1)%3    
        
    answer =[]
    for r in result:
        answer += r
    return answer